src/EventStore.Core/Settings/ThreadCountCalculator.cs

```diff
@@ -1,23 +1,28 @@
 using System;
+using EventStore.Common.Utils;
 
 namespace EventStore.Core.Settings {
 
 	public static class ThreadCountCalculator {
 		private const int ReaderThreadCountFloor = 4;
 
-		public static int CalculateReaderThreadCount(int configuredCount, int processorCount) {
+		public static int CalculateReaderThreadCount(int configuredCount, int processorCount,
+			bool isRunningInContainer) {
 			if (configuredCount > 0)
 				return configuredCount;
 
-			var readerCount = Math.Clamp(processorCount * 2, ReaderThreadCountFloor, 16);
+			if (isRunningInContainer) return ContainerizedEnvironment.ContainerReaderThreadCount;
```

Comment:
maybe a bit picky, but i think it's best to have the return on its own line to be consistent with the rest of the method